java/Nikiforov-Oleksandr/src/com/zoolatech/lecture4/tasks/_2/BinarySearchTree.java

```diff
@@ -0,0 +1,47 @@
+package com.zoolatech.lecture4.tasks._2;
+
+public class BinarySearchTree {
+    static class Node {
+        int key;
+        Node left, right;
+
+        public Node(int key) {
+            this.key = key;
+            left = right = null;
```

Comment:
left and right will have a `null` value by default, so it's not necessary to init them here